Userland/Libraries/LibCore/Stream.h

```diff
@@ -727,15 +727,14 @@ class BufferedHelper {
         // remove from the buffer after copying up to the delimiter to the
         // user buffer.
 
-        auto const find_candidates = [this, &candidates, &longest_candidate](Optional<size_t> max_offset = {}) -> Optional<OffsetAndSize> {
+        auto const find_candidates = [this, &candidates, &longest_candidate](size_t min_offset, Optional<size_t> max_offset = {}) -> Optional<OffsetAndSize> {
+            auto const corrected_minimum_offset = *longest_candidate > min_offset ? 0 : min_offset - *longest_candidate;
             max_offset = max_offset.value_or(m_buffer.used_space());
 
             Optional<size_t> longest_match;
             size_t match_size = 0;
             for (auto& candidate : candidates) {
-                // FIXME: This currently searches through the buffer from the start,
-                //        even if we just appended a small number of bytes at the end.
```

Comment:
Note that the original intention behind this `FIXME` was that we already searched through most of the buffer on a previous iteration and now only need to search the bytes that were freshly appended by the former `populate_read_buffer` (plus a bit of overlap for long patterns). I guess this works fine as well, but we need to make sure that we correctly handle overlapping patterns and that the order of the patterns doesn't have an effect on the result.

Reply:
I don't understand what you are expecting me to do.
I think that solves the overlapping issue.
```cpp
auto const corrected_minimum_offset = *longest_candidate > min_offset ? 0 : min_offset - *longest_candidate;
[...]
for (auto& candidate : candidates) {
    auto const result = m_buffer.offset_of(candidate, corrected_minimum_offset, *max_offset);
```
And AFAICT, the code that matters for pattern's order is directly copied from the old implementation.

Reply:
I just wanted to bring to your attention that what you did here is beyond the scope of what I originally intended with that FIXME. As long as everything works as intended there is no need for you to do anything, it just seemed like a slightly larger attack surface for edge-cases that was worth mentioning.

Reply:
Oh okay, for sure I didn't understand then.